python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,730 @@
+                ``num_edges_per_type`` list,
+        * ``edges`` should have the same no. of keys are the no. of edge types
+                present in the ``edge_type`` list,
+        * ``node_data``, if present, can contain keys whose count is less than
+                or equal to the node types present in the ``node_type`` list,
+                Each node type in this dictionary is expected to non-zero
+                features,
+        * ``edge_data``, if present, follows the same rules as described for
+                ``node_data`` in the earlier bullet point.
+
+    Examples:
+    ---------
+    Creating and initializing the an instance of this class during the pipelines
+    initialization process.
+    >>> base_dir = "/home/ubuntu/graph-dataset"
+    >>> file_path = "metadata.json"
+    >>> metadata_obj = Metadata(file_name, base_dir)
+
+    Once the object of this class is created it can be stored for retrieving any
+    information from the Metadata file when needed.
+    >>> graph_ntypes = metadata_obj.ntypes
+    >>> print(graph_ntypes)
+    >>> ["ntype1", "ntype2"]
+
+    In the above code snipped ``ntypes`` property is invoked to get a list of
+    node types present in the input graph, whose return value is a list of
+    strings. The above code snippet assumes that the input graph has two
+    node types namely ``ntype1`` and ``ntype2``.
+    """
+
+    def __init__(self, file_name, base_dir, init_maps=True):
+        self._basedir = None
+        self._metadata_path = None
+        self._data = None
+        self._global_nid_offsets = None
+        self._global_eid_offsets = None
+        self._ntypes = None
+        self._ntype_id_map = None
+        self._id_ntype_map = None
+        self._etype_id_map = None
+        self._id_etype_map = None
+        self._etypes = None
+        self._ntype_features = None
+        self._etype_features = None
+        self._ntype_feature_files = None
+        self._etype_feature_files = None
+        self._etype_files = None
+
+        self.__load_json(file_name, base_dir)
+        if init_maps:
+            self.__init_maps()
+            self.__validate_metadata()
+
+    ############################## Private Methods #############################
+
+    def __check_ifexists(self, data_files, file_type):
+        """Helper function to verify the presence of a set of filenames.
+
+        Argument
+        --------
+        data_files: list of str
+            A set of file names, either in absolute or in relative formats.
+        src_type: str
+            Either node type or edge type from the input graph.
+        """
+        for filename in data_files:
+            if not os.path.isabs(filename):
+                if not os.path.isfile(os.path.join(self._basedir, filename)):
+                    raise FileNotFoundError(
+                        f"File: {os.path.join(self._basedir, filename)} "
+                        f"does not exist for {file_type}"
+                    )
```

Comment:
you can just use assert to make the code shorter

Reply:
The way I understand "assert" is to ensure smooth functioning of the code and validate conditions which are "expected" to be true at some point with the application logic. I think throwing AssertionError when actually a more meaningful/appropriate errors like FileNotFoundError or KeyError can be raised is very in-appropriate. 

In favor of saving time for the CI process, I changed all the error conditions to use assert statements throughout the file.